src/QMCWaveFunctions/TWFPrototype.h

```diff
@@ -0,0 +1,223 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2021 QMCPACK developers.
+//
+// File developed by:   Raymond Clay III, rclay@sandia.gov, Sandia National Laboratories
+//
+// File created by:   Raymond Clay III, rclay@sandia.gov, Sandia National Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+/**@file TWFPrototype.h
+ *@brief Declaration of TWFPrototype
+ */
+#ifndef QMCPLUSPLUS_TWFPROTOTYPE_H
+#define QMCPLUSPLUS_TWFPROTOTYPE_H
+
+#include "QMCWaveFunctions/WaveFunctionComponent.h"
+#include "QMCWaveFunctions/SPOSet.h"
+#include "Configuration.h"
+#include "Particle/ParticleSet.h"
+namespace qmcplusplus
+{
+/**
+ *  TWFPrototype is a wrapper class for TrialWavefunction that provides separate and low level access to the Jastrow and 
+ *  SPOSet objects.  This is so that observables can be recast in matrix form and their derivatives taken efficiently. 
+ *  Currently this is hard coded for ground state slater jastrow wave functions, but generalization to
+ *  arbitrary occupations and multideterminants are straightforward and will come online as the code is tested and validated. 
+ *
+ *  Please see : J. Chem. Phys. 144, 194105 (2016) https://doi.org/10.1063/1.4948778 for implementation details and formalism.
+ */
+class TWFPrototype
+{
+public:
+  using ValueMatrix_t = SPOSet::ValueMatrix_t;
+  using GradMatrix_t  = SPOSet::GradMatrix_t;
+  using HessMatrix_t  = SPOSet::HessMatrix_t;
+  using IndexType     = QMCTraits::IndexType;
+  using RealType      = QMCTraits::RealType;
+  using ValueType     = QMCTraits::ValueType;
+
+  using ValueVector_t = SPOSet::ValueVector_t;
+  using GradVector_t  = SPOSet::GradVector_t;
+
+  TWFPrototype();
+  void add_determinant(const ParticleSet& P, const IndexType groupid, SPOSet* spo);
+  inline void add_jastrow(WaveFunctionComponent* j) { J.push_back(j); };
```

Comment:
This is new code it must follow the coding conventions for naming methods.

Reply:
Done.